tests/test_basic.py

```diff
@@ -0,0 +1,151 @@
+        packets = self._packets
+        protocol = self._protocol
+        (product, group, test, config, env) = self._test
+
+        if isinstance(python_output, str):
+            python_output = python_output.split('\n')
+
+        while(python_output[-1] == ''):
+            del python_output[-1]
+        
+        # Check for any xC device errors
+        for line in (xc_output + python_output):
+            if re.match('.*ERROR|.*error|.*Error|.*Problem', line):
+                self.record_failure(line)
+
+        # Check the board is listening on every port
+        found_connections = 0
+        for i in range(len(xc_output)):
+            if(re.match('Listening on port: [0-9]+$', xc_output[i])):
+                found_connections += 1
+
+        if found_connections < total_connections:
+            self.record_failure("Incorrect amount of listening ports created on device\n" +
+                                "  Expected ports: {}".format(total_connections) +
+                                "  Actual ports:   {}".format(found_connections))
+
+        udp_max_error = int(packets * total_connections * 0.01)
+        total_errors = -1
+        for i in range(len(python_output)):
+            if(re.match('Lost_packets: [0-9]+$', python_output[i])):
+                total_errors = int(python_output[i].split(' ')[1])
+                break
+
+        if total_errors == -1:
+            self.record_failure("Could not find the python script's error output")
+
+        # Accept a 1% error rate for UDP?
+        if protocol == "UDP":
+            if total_errors > udp_max_error:
+                self.record_failure("Packet error rate too high \n" +
+                                    "  Allowable loss rate: {} packets\n".format(udp_max_error) +
+                                    "  Actual loss rate:    {} packets\n".format(total_errors))
+
+        # Accept no errors for TCP
+        else:
+            if total_errors > 0:
+                self.record_failure("Packet error rate too high \n" +
+                                    "  Allowable loss rate: 0 packets\n" +
+                                    "  Actual loss rate:    {} packets\n".format(total_errors))
+        
+        output = {'python_output':''.join(python_output),
+                  'device_output':''.join(xc_output)}
+
+        if not self.result:
+            output['failures'] = ''.join(self.failures)
+
+        xmostest.set_test_result(product,
+                                 group,
+                                 test,
+                                 config,
+                                 self.result,
+                                 env={},
+                                 output=output)
+
+
+def test(remote_processes, connections, protocol, packets, delay):
+    setup = '{}_{}_{}'.format(remote_processes, connections, protocol)
+    binary = 'xtcp_bombard/bin/{}/xtcp_bombard_{}.xe'.format(setup, setup)
+
+    expect_filename = setup + '.expected'
```

Comment:
The expect files don't appear to be included in this pull request?
Minor, but the filename extension is normally `.expect` for these files.